crates/driver/src/domain/competition/solution/scoring.rs

```diff
@@ -194,10 +190,16 @@ impl Trade {
             order::FeePolicy::Volume { factor } => Ok(self.volume_fee(*factor)?.amount),
         };
 
-        let protocol_fee = self.policies.first().map(protocol_fee).transpose();
+        let amount = self
+            .policies
+            .iter()
+            .try_fold(TokenAmount::default(), |acc, protocol_fee| {
+                acc.checked_add(&calc_protocol_fee(protocol_fee).ok()?)
```

Comment:
I don't think it won't work this way. After each calculated protocol fee, `custom_prices` change, and `custom_prices` depend on the `executed` and `fee` fields of the fulfilment, which again change after each applied protocol fee.